### PR TITLE
new grey matter configuration style

### PR DESCRIPTION
**Pull request by alecholmez**
This change introduces a new style of CUE configuration that offers high extensibility through a CUE unification pipeline. Services now have simplified definitions with blocks of configuration designed to make understanding your mesh graph super easy. A high level EXPORTS file now contains finalized exportable objects that can be used to quickly evaluate your Grey Matter mesh configurations as well as group blocks of config together. This facilitates powerful GITOPS workflows where individual teams manage their own sync containers on a per application basis. It is recommended that the greymatter-cue and envoyproxy.io CUE definition files are imported as submodules so versioning can be guaranteed across the project. There are example scripts provided which demonstrate how this can be managed.

### CODE REVIEW

File: scripts/build [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+case $ARG in
+    help|fmt)
+        cmd_$ARG $@
+        ;;
+    *)
+        echo "unknown arg: $ARG"
+        cmd_help
+        exit 1
+        ;;
+esac
```
Comment (dontlaugh) [inline scripts/build:34]: newline

File: README.md [hunk 2]
```diff
@@ -1,149 +1,64 @@
...
-## Repo Stats
+## Project Layout
 
-Count lines of CUE
+* `cue.mod`: a directory that marks this repo as a cue module. This dir is
+  managed by the `cue` CLI.
+* `cue.mod/module.cue`: contains our module name: "produce.local". We use this
+  name to import packages, for instance the `gm` package.
+* `cue.mod/pkg`: a package that holds all the Grey Matter and Envoy config schemas in CUE.
+* `services/inputs.cue`: a cue file in `package mesh` that contains defaults, overrides, and user inputted values.
```
Comment (moosecanswim) [inline README.md:36]: Should this be `package services`?

**Comment by moosecanswim** [comment]
Hey Alec, I think this is a great rework of the cue files.  

I made some inline comments for the readme but those wont affect the logic of this repo.
- I think we should link add a link to the readme to our docs site where we explain how to apply the mesh configs. 
- Additionally i think we should add a tutorial about how to create your own mesh configs. ex:

```markdown
### How to create your own mesh configs
When you want to create your own mesh configs create a new folder in services.
Inside that directory create a file `myservice.cue` and add `package services` as the first line.
...
```
- Could you add something to the readme about what `apple/apple.cue` and`applications/*.cue`.  Why are there two cue files in applications directory and why there is only one in apple.  Stating that there is one file per data plane connection could be useful.

Question, how do we expect users to use this, from a greymatter admin perspective, as well as a tenant team?
From what i see so far is a team that owns the mesh would copy this repo and make modifications to the `inputs.cue` and `intermediate.cue`.  For mesh's that have a monorepo of mesh configs this would not be a concern but when individual teams own their mesh config repo the admin team would need to pass those files to every tenant team that will create their own mesh configs.

**Review by moosecanswim — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -1,149 +1,64 @@
...
+* `services/intermediates.cue`: a cue file in `package mesh` that contains predefined mesh object templates that services 
+  can override and use in their configuration files.
+* `services/greymatter`: core Grey Matter service configurations 
+* `services`:a directory of example service configs, and our concrete values. These
+  configs are all known to work with Grey Matter `v1.7.0+`.
+* `EXPORT.cue`: the end all final conglomerate of your mesh configurations in their respective groupings. These is what is meant
```
[the remark this answers is not in the document]
Comment (alecholmez) [reply README.md:42]: Ah so the evaluated part is very CUE specific. CUE refers to unifying values as `evaluating` so when we use the CUE CLI like `cue eval -e all_configs` we are saying evaluate the result of `all_configs`. I think yea I can massage that wording to make that a bit more understandable.

File: EXPORT.cue [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+// You may specify which array of configs you'd like to sync or bundle them all in one.
+// We recommend splitting out your configs through namespaces such as core services,
+// business applications, etc...
+
+// example evaluation commands:
+// cue eval -c EXPORT.cue --out json -e greymatter_configs
+// cue eval -c EXPORT.cue --out yaml -e service_configs
+// cue eval -c EXPORT.cue --out json -e all_configs
+
+package produce
```
[the remark this answers is not in the document]
Comment (alecholmez) [reply EXPORT.cue:16]: Okay cool, I will add some comments around this. My idea behind this package name was it was a mesh name. So for example the repo might hold SEED mesh configs, so we can call the final package `seed`.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (moosecanswim) [reply README.md:42]: Thanks for the clarification.

File: EXPORT.cue (hunk 4 above)
[the remark this answers is not in the document]
Comment (moosecanswim) [reply EXPORT.cue:16]: Ahhh icic.  By change `seed` somewhat works with `produce`

**Comment by alecholmez** [comment]
Kyle's questions were answered internally at this meeting: [notes](https://docs.google.com/document/d/1IDlKSkWc5PfTA80_tgU2XGUM8RdMtUYo8tdupKDC33w/edit#heading=h.29o1v4eet16n)

**Pull request merged**